File: Inventarios.Api.Nest/productos.api/src/application/helpers/password.helper.ts
/**
 * Helper estático para validaciones de contraseña.
 * Mismo concepto que en Inventarios.Api (.NET)
 */
export class PasswordHelper {
  private static readonly specialCharacters = '!@#$%^&*()_+-=[]{}|;\':",./<>?';

  static hasValidComplexity(password: string): boolean {
    return (
      /[A-Z]/.test(password) &&
      /[a-z]/.test(password) &&
      /\d/.test(password) &&
      [...password].some(c => PasswordHelper.specialCharacters.includes(c))
    );
  }

  static hasMinimumLength(password: string, minLength = 8): boolean {
    return password.length >= minLength;
  }

  static hasMaximumLength(password: string, maxLength = 128): boolean {
    return password.length <= maxLength;
  }
}

File: Inventarios.Api/Inventarios.api.Application/Helpers/PasswordHelper.cs
namespace Inventarios.api.Application.Helpers;

public static class PasswordHelper
{
    private const string SpecialCharacters = "!@#$%^&*()_+-=[]{}|;':\",./<>?";

    public static bool HasValidComplexity(string password) =>
        password.Any(char.IsUpper) &&
        password.Any(char.IsLower) &&
        password.Any(char.IsDigit) &&
        password.Any(c => SpecialCharacters.Contains(c));

    public static bool HasMinimumLength(string password, int minLength = 8) =>
        password.Length >= minLength;

    public static bool HasMaximumLength(string password, int maxLength = 128) =>
        password.Length <= maxLength;
}